### PR TITLE
Improve visitor execution logic

### PR DESCRIPTION
**Pull request by ThisaruGuruge**
## Purpose
$Subject

Fixes: [#1511](https://github.com/ballerina-platform/ballerina-standard-library/issues/1511)

## Examples
N/A

## Checklist
- [x] Linked to an issue
- [ ] ~Updated the changelog~
- [ ] ~Added tests~
- [ ] ~Updated the spec~

### CODE REVIEW

File: ballerina/engine.bal [hunk 1]
```diff
@@ -46,10 +48,24 @@ isolated class Engine {
         }
     }
 
-    isolated function execute(parser:OperationNode operationNode, Context context, map<Upload|Upload[]> fileInfo)
-        returns OutputObject {
-        ExecutorVisitor executor = new (self, self.schema, context, fileInfo);
-        OutputObject outputObject = executor.getExecutorResult(operationNode);
+    isolated function getResult(parser:OperationNode operationNode, Context context, map<Upload|Upload[]> fileInfo,
+        any result = ()) returns OutputObject {
```
Comment (DimuthuMadushan) [inline ballerina/engine.bal:52]:
```suggestion
    isolated function getResult(parser:OperationNode operationNode, Context context, map<Upload|Upload[]> fileInfo,
                                any result = ()) returns OutputObject {
```

File: ballerina/executor_visitor.bal [hunk 2]
```diff
@@ -27,7 +27,9 @@ class ExecutorVisitor {
     private map<Upload|Upload[]> fileInfo;
     private any result;
 
-    isolated function init(Engine engine, __Schema schema, Context context, map<Upload|Upload[]> fileInfo, any result = ()) {
+    isolated function init(Engine engine, __Schema schema, Context context, map<Upload|Upload[]> fileInfo,
+        any result = ()) {
```
Comment (DimuthuMadushan) [inline ballerina/executor_visitor.bal:31]:
```suggestion
    isolated function init(Engine engine, __Schema schema, Context context, map<Upload|Upload[]> fileInfo,
                           any result = ()) {
```

File: ballerina/field_validator_visitor.bal [hunk 3]
```diff
@@ -0,0 +1,662 @@
...
+            self.validateListVariableValue(variableValue, schemaArg, argumentNode.getValueLocation(), fieldName);
+            self.removePath();
+        } else if variableValue is () {
+            self.validateArgumentValue(variableValue, argumentNode.getValueLocation(), getTypeName(argumentNode), schemaArg);
+        } else {
+            string expectedTypeName = getOfTypeName(schemaArg.'type);
+            string listError = getListElementError(self.argumentPath);
+            string value = variableValue is () ? "null" : variableValue.toString();
+            string message = string`${listError}${expectedTypeName} cannot represent non ${expectedTypeName} value:` +
+                             string` ${value}`;
```
Comment (DimuthuMadushan) [inline ballerina/field_validator_visitor.bal:209]:
```suggestion
            string message = string `${listError}${expectedTypeName} cannot represent non ${expectedTypeName} value:` +
                             string ` ${value}`;
```

File: ballerina/engine.bal (hunk 1 above)
Comment (ThisaruGuruge) [reply ballerina/engine.bal:52]: This is the way the Ballerina style guide recommends using indentations. (There are places where we do not follow this, we should check them later).

File: ballerina/executor_visitor.bal (hunk 2 above)
Comment (ThisaruGuruge) [reply ballerina/executor_visitor.bal:31]: Same as above.

**Review by DimuthuMadushan — APPROVED** [review]

**Pull request merged**